client/my-sites/plugins/main.jsx

```diff
@@ -189,9 +197,17 @@ export default React.createClass( {
 		);
 	},
 
-	toggleBulkManagement() {
-		const bulkManagement = ! this.state.bulkManagement,
-			sites = bulkManagement ? this.props.sites.getSelectedOrAllWithPlugins() : this.props.sites.getSelectedOrAll();
+	toggleBulkManagement( section ) {
+		let bulkManagement = this.state.bulkManagement;
+		const foundIndex = bulkManagement.indexOf( section );
+
+		if ( foundIndex === -1 ) {
+			bulkManagement.push( section );
+		} else {
+			bulkManagement.splice( foundIndex, 1 );
+		}
```

Comment:
There's another possible way to accomplish this that might be slightly slower but hopefully more readable…

```js
import negate from 'lodash/function/negate';
const equals = a => b => a === b;

toggleBulkManagement( section ) {
	let { bulkManagement } = this.state;

	bulkManagement = bulkManagement.some( equals( section ) )
		? bulkManagement.filter( negate( equals( section ) ) // remove if present
		: bulkManagement.concat( section ); // add if missing

	…

	this.setState( {
		plugins: this.getPluginsFromStore( this.props, sites ),
		filter: 'all',
		bulkManagement
	} );
}
```